File: admin-backend/admin-system/src/main/java/io/github/faustofan/admin/system/infrastructure/reponsitory/SysOrgRepository.java
package io.github.faustofan.admin.system.infrastructure.reponsitory;

import io.github.faustofan.admin.system.domain.model.SysOrg;
import io.github.faustofan.admin.system.domain.model.SysOrgTable;
import io.github.faustofan.admin.system.domain.model.Tables;
import org.babyfish.jimmer.spring.repository.JRepository;
import org.jspecify.annotations.NonNull;
import org.springframework.stereotype.Repository;

/**
 * 系统组织机构仓库
 * 提供对 SysOrg 实体的数据库操作方法
 */
@Repository
public interface SysOrgRepository extends JRepository<SysOrg, Long> {

    SysOrgTable table = Tables.SYS_ORG_TABLE;

    /**
     * 检查组织机构是否存在
     *
     * @param id 组织机构ID
     * @return 如果组织机构存在则返回 true，否则返回 false
     */
    boolean existsById(@NonNull Long id);
}

File: src/main/kotlin/io/github/faustofan/admin/system/infrastructure/persistence/SysOrgRepository.kt
package io.github.faustofan.admin.system.infrastructure.persistence

import io.github.faustofan.admin.system.domain.model.SysOrg
import org.babyfish.jimmer.spring.repository.KRepository
import org.springframework.stereotype.Repository

@Repository
interface SysOrgRepository : KRepository<SysOrg, Long> {

    // 查询根节点 (父节点为空)
//    fun findByParentIsNull(): List<SysOrg>

    // 根据名称模糊搜索
//    fun findByNameLike(name: String): List<SysOrg>

    // 注意：获取子部门通常不需要在这里写方法，
    // 而是通过 Entity 里的 @OneToMany children 属性配合 Fetcher 直接抓取
}